staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -107,6 +107,9 @@ func SerializeObject(mediaType string, encoder runtime.Encoder, hw http.Response
 		contentEncoding: negotiateContentEncoding(req),
 		hw:              hw,
 		ctx:             ctx,
+		// A minimal LIST JSON body has 80 bytes:
+		// {"kind":"List","apiVersion":"v1","metadata":{"resourceVersion":"1"},"items":[]}\n
+		buffer: make([]byte, 0, 80),
```

Comment:
this always allocates a buffer, even for responses that would never need it (because they aren't using gzip or the first write would exceed the minimum gzip size)